engine/class_modules/sc_death_knight.cpp

```diff
@@ -7275,14 +7312,21 @@ void death_knight_t::init_spells()
   spec.frost_death_knight  = find_specialization_spell( "Frost Death Knight" );
   spec.frost_fever         = find_specialization_spell( "Frost Fever" );
   spec.frost_strike        = find_specialization_spell( "Frost Strike" );
+  spec.frost_strike_rank_2 = find_specialization_spell( 316803 );
   spec.howling_blast       = find_specialization_spell( "Howling Blast" );
   spec.obliterate          = find_specialization_spell( "Obliterate" );
+  spec.obliterate_rank_2   = find_specialization_spell( 317198 );
   spec.rime                = find_specialization_spell( "Rime" );
+  spec.rime_rank_2         = find_specialization_spell( 316838 );
   spec.runic_empowerment   = find_specialization_spell( "Runic Empowerment" );
   spec.killing_machine     = find_specialization_spell( "Killing Machine" );
+  spec.killing_machine_rank_2     = find_specialization_spell( 317214 );
   spec.empower_rune_weapon = find_specialization_spell( "Empower Rune Weapon" );
+  spec.empower_rune_weapon_rank_2 = find_specialization_spell( 317230 );
   spec.pillar_of_frost     = find_specialization_spell( "Pillar of Frost" );
+  spec.pillar_of_frost_rank_2    = find_specialization_spell( 316849 );
   spec.remorseless_winter  = find_specialization_spell( "Remorseless Winter" );
+  spec.remorseless_winter_rank_2 = find_specialization_spell( 316794 );
```

Comment:
You should be able to use find specialization_spell( name, token) to get spell ranks without looking for specific IDs.

example used in unholy: ``  spec.raise_dead_2        = find_specialization_spell( "Raise Dead", "Rank 2" );``